src/Soneta.Sdk/Sdk/Sdk.targets

```diff
@@ -18,6 +26,152 @@
     <RunArguments>$(StartArguments)</RunArguments>
   </PropertyGroup>
 
+  <UsingTask TaskName="CheckForSdkUpdates" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildBinPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <GlobalJsonFilePath ParameterType="System.String" Required="true" />
+      <LatestVersionFilePath ParameterType="System.String" Required="true" />
+      <NugetApiUrl ParameterType="System.String" Required="true" />
+      <UpdateInterval ParameterType="System.Int32" Required="true" />
+      <UpdateIntervalUnit ParameterType="System.String" Required="true" />
+      <IsUpdateAvailable ParameterType="System.Boolean" Output="true" />
+      <CurrentVersion ParameterType="System.String" Output="true" />
+      <LatestVersion ParameterType="System.String" Output="true" />
+    </ParameterGroup>
+
+    <Task>
+      <Using Namespace="System.Linq" />
+      <Using Namespace="System.Net.Http" />
+      <Using Namespace="System.Text.RegularExpressions" />
+      <Code Type="Method" Language="cs">
+        <![CDATA[
+        public enum TimeUnit
+        {
+            None,
+            Seconds,
+            Minutes,
+            Hours,
+            Days,
+            Weeks,
+            Months,
+            Years
+        }
+
+        private const string GlobalJsonSdkVersionPattern = @"(?<=""Soneta.Sdk""\s*:\s*"")([0-9]+\.?){3,}(?=.*?"")";
+        private const string NugetApiLatestVersionPattern = @"(?<="")([0-9]+\.?){3,}(?=.*?""\s*])";
+
+        public override bool Execute()
+        {
+            CurrentVersion = GetCurrentVersion();
+            LatestVersion = GetLatestVersion();
+            IsUpdateAvailable = CheckIfUpdateAvailable(CurrentVersion, LatestVersion);
+
+            return true;
+        }
+
+        private string GetCurrentVersion()
+        {
+            if (File.Exists(GlobalJsonFilePath))
+            {
+                var json = File.ReadAllText(GlobalJsonFilePath);
+                return GetMatchingValue(json, GlobalJsonSdkVersionPattern);
+            }
+
+            return null;
+        }
+
+        private string GetMatchingValue(string json, string pattern)
+        {
+            var match = Regex.Match(json, pattern);
+            return match.Value;
+        }
+
+        private string GetLatestVersion()
+        {
+            return CalculateUpdateCheckTime() < DateTime.Now ? 
+                GetRemoteVersion() : GetLocalVersion();
+        }
+
+        private DateTime CalculateUpdateCheckTime() 
+        {
+            var lastWriteTime = File.GetLastWriteTime(LatestVersionFilePath);
```

Comment:
@swiato czy nie warto sprawdzić czy ten plik istnieje podobnie jak zrobiłeś to w przypadku global.json?

Reply:
O ile się nie mylę, metoda File.GetLastWriteTime nie rzuca wyjątku w przypadku braku pliku, tylko zwraca wartość DateTime.Min. Takie zachowanie sprawia, że osiągamy pożądane zachowanie, lastWriteTime jest wtedy mniejsze od DateTime.Now i wersja zostaje pobrana z serwera i utrwalona w nowym pliku. Faktycznie, nie jest to oczywiste patrząc na kod, pozostawiam do dyskusji temat, czy należy tu sprawić jawne sprawdzenie obecności pliku ;)